resources/session02/homework/http_server.py

```diff
@@ -17,25 +35,69 @@ def response_method_not_allowed():
     resp = []
     resp.append("HTTP/1.1 405 Method Not Allowed")
     resp.append("")
+    resp.append("")
     return "\r\n".join(resp).encode('utf8')
 
 
 def response_not_found():
     """returns a 404 Not Found response"""
-    return b""
+    resp = []
+    resp.append("HTTP/1.1 404 Not Found")
+    resp.append("")
+    resp.append("")
+    return "\r\n".join(resp).encode('utf8')
+
+
+def server_error():
+    """returns a 500 Server response"""
+    resp = []
+    resp.append("HTTP/1.1 500 Server Error")
+    resp.append("")
+    resp.append("")
+    return "\r\n".join(resp).encode('utf8')
 
 
 def parse_request(request):
     first_line = request.split("\r\n", 1)[0]
+    print('Request: "{}"'.format(first_line))
     method, uri, protocol = first_line.split()
     if method != "GET":
         raise NotImplementedError("We only accept GET")
     return uri
 
 
+def run_external_script(uri):
+    p = Popen([uri], stdout=PIPE)
+    return p.stdout.read()
+
+
 def resolve_uri(uri):
-    """This method should return appropriate content and a mime type"""
-    return b"still broken", b"text/plain"
+    """Return appropriate content and a mime type"""
+    uri = re.sub('\.\.', '', uri)  # pull any updirs out of the uri
```

Comment:
This is a good security thought.  You might want to consider though if this is really what you want to do?  Relative uris with climbing to parent dirs are common in css and javascript paths and simply erasing them might be confusing.  

Given, of course, that this is a very simplistic HTTP server, it's probably not a big deal, and it's very good that you caught this possible weakness.  Just consider what other options you might have.